### PR TITLE
Milestone: Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by Lancelot-SO**
### The following updates were made:

- The application includes several models based on an Entity-Relationship Diagram (ERD) in PostgresSQL. The models include
- User, Group, Expense, and a bridge table called GroupExpense.
- Authentication is implemented using Devise, while authorization is handled by Cancancan.
- An unauthorized user is presented with a splash screen upon accessing the application.
- The category page serves as the home page for the application.
- A transaction page has been created.
- Unit and integration tests have been written using RSpec.
- Authenticated users have the ability to create groups and expenses.
- A loom video presentation showcasing the project's details.
- Project deployed on the Render site
- The design of the application adhered to a mobile-first approach
- Linters setup properly.
- Modified Readme file

### CODE REVIEW

**Review by youmari — COMMENTED** [review]
Hi @Lancelot-SO , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that follows git-flow you should open a PR from a feature branch to the dev branch that way you will be followed the git flow 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Lancelot-SO** [comment]
The project follows git flow and as such it can be done this way

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  validates :name, presence: true
```
Comment (Ismailco) [inline app/models/user.rb:6]: - [ ] Kindly make sure to validate all the user inputs, this includes password and email as well as the name

File: README.md [hunk 2]
```diff
@@ -1 +1,151 @@
...
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+- <p> I would like to thank microverse community for the support and guidance</p>
+- <p> I would like to express gratitude to <a href="https://www.behance.net/gregoirevella">Gregoire Vella</a>, as per the Creative Commons license.</p>
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (Ismailco) [inline README.md:149]: - [ ] Kindly make sure to add a license to your repo and use the correct link here, as for now the license link redirect to a 404 page

**Review by Ismailco — CHANGES_REQUESTED** [review]
#### Hi @Lancelot-SO, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Suggested changes
- [ ] Kindly make sure your app has unit tests for all the most important components of the RoR application
- [ ] Kindly consider adding integration tests for all the most important components in your app so it's well tested and covers all the use cases in the app. also make sure to add validation tests

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/models/user.rb (hunk 1 above)
Comment (Lancelot-SO) [reply app/models/user.rb:6]: Please all inputs are validated. The name has also been validated as can be seen in the user model

Comment (Ismailco) [reply app/models/user.rb:6]: @Lancelot-SO, I can only see that the name is validated in the user model, you should add validation for the password and email also

Comment (Lancelot-SO) [reply app/models/user.rb:6]: The validation for password and email is comes along with setting up devise 
The password and email are validated 
you can't access the app without proper sign up credentials

Comment (Lancelot-SO) [reply app/models/user.rb:6]: Also, for all past projects that I have done, this is how the validations are done. After i set up devise i get both password and email validated

**Review by Danieldotcomcoder — CHANGES_REQUESTED** [review]
Hello @Lancelot-SO , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights
Professional Readme file ✔️✔️
Clear and nice code ✔️✔️

## Required Changes ♻️

_ As mentioned by the previous reviewer, kindly make sure to add unit and integration tests for all tha major component as it's very important and mentioned in the project requirments.

As you can notice, there are not tests right now: 
image

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by youmari — APPROVED** [review]
Hi @Lancelot-SO , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights
✔️ PRO PR
✔️ No linter errors

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**